### PR TITLE
SQL-1140: Update ODBC Driver name in Power BI connector

### PR DESCRIPTION
**Pull request by bucaojit**
Updated `ADF_ODBC_DRIVER` to `MONGODB_ODBC_DRIVER`

### CODE REVIEW

File: connector/MongoDBAtlasODBC.pq [hunk 1]
```diff
@@ -111,7 +111,7 @@ shared GetConnectionString = (uri as text, database as text) as record =>
     let
         DatabaseNotSet = (database = ""),
         ConnectionString = [
-            DRIVER = "ADF_ODBC_DRIVER",
+            DRIVER = "MONGODB_ODBC_DRIVER",
```
Comment (nbagnard) [inline connector/MongoDBAtlasODBC.pq:114]: umh, this is what you'll see in ODBC administrator. Does not look much better. Let's ping Alexi again.

**Review by terakilobyte — APPROVED** [review]

File: connector/MongoDBAtlasODBC.pq (hunk 1 above)
Comment (bucaojit) [reply connector/MongoDBAtlasODBC.pq]: Updated the name to `MongoDB Atlas SQL ODBC Driver`

**Review by nbagnard — APPROVED** [review]
LGTM

**Comment by bucaojit** [comment]
evergreen merge

**Pull request merged**